src/sources/internal_metrics.rs

```diff
@@ -9,23 +9,12 @@ use futures::{stream, SinkExt, StreamExt};
 use serde::{Deserialize, Serialize};
 use tokio::time;
 
-#[serde(deny_unknown_fields)]
-#[derive(Deserialize, Serialize, Debug, Clone)]
+#[derive(Deserialize, Serialize, Debug, Clone, Derivative)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields, default)]
 pub struct InternalMetricsConfig {
-    #[serde(default = "default_scrape_interval_secs")]
```

Comment:
Did you find why this wasn't being applied? Just wondering so we can avoid this situation.

Reply:
I didn't find the exact cause, but I suspect the cause was the lack of `default` in serde macro.

https://github.com/timberio/vector/blob/491302313e3e60839bf5ce962ef20302c4caa3d4/src/sources/internal_metrics.rs#L14

Reply:
Hrm, that _shouldn't_ be the case from my understanding. Putting `default` on the container causes it to use the `Default` implementation for the container, but I don't think that is required for using `default` on individual fields.

For example, it worked fine on the `apache_metrics` source:

https://github.com/timberio/vector/blob/ef45085f1c217fea58c9c608314baeae596945e2/src/sources/apache_metrics/mod.rs#L27-L34

I feel like something else must have been going on here.

Reply:
The default actually seems fine on `master` for me:

```
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:18.317974885Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
{"name":"uptime_seconds","timestamp":"2020-12-23T16:24:20.317064449Z","tags":null,"kind":"absolute","gauge":{"value":2.0}}
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:20.317135170Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
{"name":"uptime_seconds","timestamp":"2020-12-23T16:24:22.317164079Z","tags":null,"kind":"absolute","gauge":{"value":4.0}}
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:22.317218451Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
{"name":"uptime_seconds","timestamp":"2020-12-23T16:24:24.316297392Z","tags":null,"kind":"absolute","gauge":{"value":6.0}}
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:24.316355289Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
{"name":"uptime_seconds","timestamp":"2020-12-23T16:24:26.316387415Z","tags":null,"kind":"absolute","gauge":{"value":8.0}}
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:26.316443145Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
{"name":"uptime_seconds","timestamp":"2020-12-23T16:24:28.317004368Z","tags":null,"kind":"absolute","gauge":{"value":10.0}}
{"name":"vector_started_total","timestamp":"2020-12-23T16:24:28.317064219Z","tags":null,"kind":"absolute","counter":{"value":1.0}}
```

With:

```toml
[sources.in]
  type = "internal_metrics"

[sinks.out]
  type = "console"
  encoding.codec = "json"
  inputs = ["in"]
```

Am I misunderstanding the issue?

Reply:
I'm wondering if maybe the metrics collection itself, in the loop, was taking more than 2 seconds causing it to look like there was no default?

Reply:
Well, I first added a log print at the source `run` fn and checked through that - the value was `0`, so didn't bother looking any further. :laughing: 
Interesting, if it works on `master`...
I'll have to double-check everything. :+1:

Reply:
You're right, it is quite odd. We can still merge this PR though I guess - it simplifies things and adds a safety warning - still useful.